rsocket-transport-netty/src/main/java/io/rsocket/transport/netty/server/TcpServerTransport.java

```diff
@@ -47,15 +54,21 @@ public static TcpServerTransport create(TcpServer server) {
 
     @Override
     public Mono<Closeable> start(ConnectionAcceptor acceptor) {
-        return server
-            .newHandler((in, out) -> {
-                in.context().addHandler("server-length-codec", new RSocketLengthCodec());
-                NettyDuplexConnection connection = new NettyDuplexConnection(in, out, in.context());
-                acceptor.apply(connection).subscribe();
-
-                return out.neverComplete();
-            })
-            .map(NettyContextClosable::new);
+        Mono<? extends NettyContext> serverMono = server
+                .newHandler((in, out) -> {
+                    in.context().addHandler("server-length-codec", new RSocketLengthCodec());
+                    NettyDuplexConnection connection = new NettyDuplexConnection(in, out, in.context());
+                    acceptor.apply(connection).subscribe();
+
+                    return out.neverComplete();
+                });
+
+        serverMono.map(s -> s.address()).subscribe(address);
```

Comment:
I hadn't considered that you would reuse a TcpServerTransport.  It seems problematic to do so as the code currently stands as you either

1) use a static port and clients are responsible for remembering the state e.g. is there an open server port?  The start call will fail unless a client has called closed.
2) use a free port (0) and then multiple calls to start work but with the race condition and address() is on the wrong class.

Overall this makes me think my original comment in the PR is correct "Alternatively we can make sure that server is a better type than Closeable."  So this start method should return NettyDuplexConnection instead of Closeable.